fedimint-server/src/net/api.rs

```diff
@@ -296,6 +310,73 @@ impl ConsensusApi {
             .await
             .map_err(|_| ApiError::server_error("Unable send event".to_string()))
     }
+
+    pub async fn build_full_server_status(&self) -> ApiResult<FullStatus> {
+        let our_epoch = self.get_epoch_count().await;
+        // Clone to avoid holding the lock while we calculate things
+        let rejoin_at_epoch = self.rejoin_at_epoch.lock().await.clone();
+        let last_epoch_by_peer = rejoin_at_epoch.into_iter().fold(
+            HashMap::<PeerId, u64>::new(),
+            |mut acc, (epoch, peers)| {
+                let epoch = epoch + 1;
+                for peer in peers {
+                    match acc.entry(peer) {
```

Comment:
`acc.entry(peer).and_modify(...).or_insert(...)` should be neater.

https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.and_modify